Fixed a bug where debug code was included in the release, resulting i…

myan-o opened the pull request:
The release build fails.

- [x] I have read the [contributing guidelines](https://github.com/ggerganov/llama.cpp/blob/master/CONTRIBUTING.md)
- Self-reported review complexity:
  - [ ] Low
  - [ ] Medium
  - [ x] High

chraac commented on ggml/src/CMakeLists.txt:
```diff
@@ -885,7 +885,11 @@ if (GGML_QNN)
+        if (NOT DEFINED ENV{TERMUX_VERSION} OR "$ENV{TERMUX_VERSION}" STREQUAL "")
+            set(GGML_QNN_DEFAULT_LIB_SEARCH_PATH "\"/data/local/tmp/\"")
+        else()
+            set(GGML_QNN_DEFAULT_LIB_SEARCH_PATH "\"$ENV{PREFIX}/lib/\"")
```
thanks for the change here, for customized prefix, how about we turn this `GGML_QNN_DEFAULT_LIB_SEARCH_PATH` into an option, so that we can assign it from outside, for your case from termux

chraac replied:
can have something like: 
```cmake
set(GGML_QNN_DEFAULT_LIB_SEARCH_PATH "\"/data/local/tmp/\"" CACHE STRING "customized library search path for QNN backend")
```

myan-o replied:
Thank you for your review. I understand.

myan-o commented:
Changed GGML_QNN_DEFAULT_LIB_SEARCH_PATH to be set from command line arguments. Also added a process to add the / character to the last path even if it does not exist.
